packages/react-conformance/src/utils/merge.tsx

```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isObject = (o: any) => o !== null && typeof o === 'object' && !Array.isArray(o);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const _merge = (obj1: any, obj2: any) => {
+  const keys: string[] = Object.getOwnPropertyNames(obj2);
+  keys.forEach(k => {
+    if (obj1 && !obj1.hasOwnProperty(k)) {
+      obj1[k] = obj2[k];
+    } else if (isObject(obj2[k])) {
+      _merge(obj1[k], obj2[k]);
+    } else if (Array.isArray(obj2[k])) {
+      obj1[k] = Array.isArray(obj1[k]) ? obj1[k].concat(obj2[k]) : obj1[k];
+    }
+  });
+};
+
+/**
+ * Merge function with specific type
+ * @param objs - objects to be merged
+ */
+export function merge<T = Object>(...objs: Object[]): T {
+  const merged = { ...objs[0] };
+  objs = objs.slice(1);
+
+  objs.forEach(option => {
+    _merge(merged, option);
+  });
+
+  return merged as T;
```

Comment:
Just wondering why this is better than any merge function, I
e. from Lodash?

Reply:
IIRC Lodash's merge may not have handled arrays the way we needed (concatenating).

Reply:
Confirmed here: https://runkit.com/ecraig12345/5f1b54969d6aa400192f4eb7

Reply:
According to unit tests in this PR I don't see any difference as Lodash performs deep merge, too.

Reply:
Oops, I see 🙃 However, `mergeWith()` the exact thing that you want, https://lodash.com/docs/4.17.15#mergeWith. An example in docs covers this use case

Reply:
Just tried out a bunch of merge functions and `webpack-merge` is the only thing I could find with the desired array behavior. Seems weird to use something with "webpack" in the name for this generic purpose, but I guess we could? Probably better than maintaining our own.

Reply:
Oops just saw the mergeWith comment. Tracking that and the dependency change in #14205.